examples/seismic/tutorials/04_dask.ipynb

```diff
@@ -178,41 +180,40 @@
     "        'observed' data using the real model. For a real case\n",
     "        the worker would be reading seismic data from disk.\n",
     "    \"\"\"\n",
-    "    true_model = get_true_model()\n",
-    "    shot_id = param['shot_id']\n",
-    "    \n",
-    "    src_coordinates = np.empty((1, len(param['shape'])))\n",
-    "    src_coordinates[0, :] = [30, param['shot_id']*1000./(param['nshots']-1)]\n",
+    "    # Reconstruct objects\n",
+    "    with open(\"arguments.pkl\", \"rb\") as cp_file:\n",
+    "        cp = pickle.load(cp_file)\n",
+    "        \n",
```

Comment:
See the answer to your other comment below. https://github.com/opesci/devito/pull/945/files/243249b2ba50444bac7370dcd00789a6245ef90c#r331614314

Reply:
Can we do it without writing and reading a file?

Reply:
Geometry is an attribute of the solver so should only need to save solver and get it via `solver.geometry`

Reply:
> Can we do it without writing and reading a file?

Yes, but it would have to be created a new `solver` at least one more time, i.e. in the `fwi_gradient` function. The new `solver` would be passed to `fwi_gradient_i` function to compute the gradient for each shot.  As this function runs in parallel (via Dask) the `solver` would have to be sent to all workers through a `broadcast` or loaded from a file. I chose the last option. I don't know if this would work on a cluster as @mloubout pointed out.